src/app/store/slices/storage/fileUtils/checkDuplicatedFiles.ts

```diff
@@ -0,0 +1,50 @@
+import { items as itemUtils } from '@internxt/lib';
+import { DriveFileData } from '@internxt/sdk/dist/drive/storage/types';
+import newStorageService from '../../../../drive/services/new-storage.service';
+
+export interface DuplicatedFilesResult {
+  duplicatedFilesResponse: DriveFileData[];
+  filesWithDuplicates: (File | DriveFileData)[];
+  filesWithoutDuplicates: (File | DriveFileData)[];
+}
+
+export const checkDuplicatedFiles = async (
+  files: (File | DriveFileData)[],
+  parentFolderId: string,
+): Promise<DuplicatedFilesResult> => {
+  if (files.length === 0) {
+    return {
+      duplicatedFilesResponse: [],
+      filesWithDuplicates: [],
+      filesWithoutDuplicates: files,
+    } as DuplicatedFilesResult;
+  }
+
+  const filesToUploadParsedToCheck = files.map((file) => {
+    const { filename, extension } = itemUtils.getFilenameAndExt(file.name);
+    return { plainName: filename, type: extension };
+  });
+
+  const checkDuplicatedFileResponse = await newStorageService.checkDuplicatedFiles(
```

Comment:
Maybe you can extract { existentFiles } directly from here. Or cast it { existentFiles: duplicatedFilesResponse }, but I think existentFiles is enough to understand what this call returns